sim/hunter/steady_shot.go

```diff
@@ -0,0 +1,59 @@
+package hunter
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+)
+
+func (hunter *Hunter) registerSteadyShotSpell() {
+
+	ssMetrics := hunter.NewFocusMetrics(core.ActionID{SpellID: 56641})
+
+	hunter.SteadyShot = hunter.RegisterSpell(core.SpellConfig{
+		ActionID:    core.ActionID{SpellID: 56641},
+		SpellSchool: core.SpellSchoolPhysical,
+		ProcMask:    core.ProcMaskRangedSpecial,
+		Flags:       core.SpellFlagMeleeMetrics | core.SpellFlagIncludeTargetBonusDamage | core.SpellFlagAPL,
+		FocusCost: core.FocusCostOptions{
+
+			Cost: 0,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD:      time.Second,
+				CastTime: time.Second * 2,
+			},
+			IgnoreHaste: true,
+			ModifyCast: func(_ *core.Simulation, spell *core.Spell, cast *core.Cast) {
+				cast.CastTime = spell.CastTime()
+			},
+
+			CastTime: func(spell *core.Spell) time.Duration {
+				return time.Duration(float64(spell.DefaultCast.CastTime) / hunter.RangedSwingSpeed())
+			},
+		},
+
+		DamageMultiplierAdditive: 1 + core.TernaryFloat64(hunter.HasPrimeGlyph(proto.HunterPrimeGlyph_GlyphOfSteadyShot), 0.1, 0),
+		DamageMultiplier: 0.62,
+		CritMultiplier: hunter.CritMultiplier(true, false, false),
+		ThreatMultiplier: 1,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			if hunter.Talents.ImprovedSteadyShot > 0 {
+				if !hunter.ImprovedSteadyShotAuraCounter.IsActive() {
+					hunter.ImprovedSteadyShotAuraCounter.Activate(sim)
+				}
+			}
+			baseDamage := hunter.AutoAttacks.Ranged().CalculateWeaponDamage(sim, spell.RangedAttackPower(target)) + (280 + spell.RangedAttackPower(target) * 0.021)
+			focus := 9.0
+			if hunter.Talents.Termination != 0 && sim.IsExecutePhase25() {
+				focus = float64(hunter.Talents.Termination) * 3
+			}
+			hunter.AddFocus(sim, focus, ssMetrics)
+			result := spell.CalcDamage(sim, target, baseDamage, spell.OutcomeRangedHitAndCrit)
+			spell.DealDamage(sim, result)
```

Comment:
If you're not doing anything with the result you can just do `spell.CalcAndDealDamage`.

Now that I look at this lots of your ranged spells are missing travel time. You can check how to implement that from the SOD fork (I did implement them all with travel time even if the missile speed is not entirely correct)